ipa-core/src/protocol/ipa_prf/boolean_ops/share_conversion_aby.rs

```diff
@@ -126,84 +128,113 @@ where
     );
 
     // Ensure that the probability of leaking information is less than 1/(2^128).
-    debug_assert!(x.iter().count() < (BITS - 128));
-
-    // generate sh_r = (0, 0, sh_r) and sh_s = (sh_s, 0, 0)
-    // the two highest bits are set to 0 to allow carries for two additions
-    let (sh_r, sh_s) =
-        gen_sh_r_and_sh_s::<_, BITS, NC>(&ctx.narrow(&Step::GenerateSecretSharing), record_id);
+    debug_assert!(input_shares.first().iter().count() < (BITS - 128));
 
-    // generate upgraded context
-    // we expect 2*256 = 512 gates in total for two additions
+    // we expect 2*256 = 512 gates in total for two additions per conversion
     // the vectorization factor is NC
-    // the total amount of multiplications is therefore NC*512
-    // make sure proof does not get too big, i.e. less that 100k
+    // the length of input_shares is the amount of converted shares
+    // the total amount of multiplications is therefore NC*512*input_shares.len()
+    // make sure proof does not get too big,
+    // i.e. less that NC*input_shares.len()<120k
     // (such that there are less than 50m multiplications)
-    assert!(NC < 100_000);
-    let validator = ctx.dzkp_validator(NC);
-    let m_ctx = validator.context();
-
-    // addition r+s might cause carry,
-    // this is no problem since we have set bit 254 of sh_r and sh_s to 0
-    let sh_rs = {
-        let (mut rs_with_higherorderbits, _) = integer_add::<_, TwoHundredFiftySixBitOpStep, NC>(
-            m_ctx.narrow(&Step::IntegerAddBetweenMasks),
-            record_id,
-            &sh_r,
-            &sh_s,
-        )
-        .await?;
-
-        // PRSS/Multiply masks added random highest order bit,
-        // remove them to not cause overflow in second addition (which is mod 256):
-        rs_with_higherorderbits[BITS - 1] = AdditiveShare::<Boolean, NC>::ZERO;
-
-        // return rs
-        rs_with_higherorderbits
-    };
-
-    // addition x+rs, where rs=r+s might cause carry
-    // this is not a problem since bit 255 of rs is set to 0
-    let (sh_y, _) = integer_add::<_, TwoHundredFiftySixBitOpStep, NC>(
-        m_ctx.narrow(&Step::IntegerAddMaskToX),
-        record_id,
-        &sh_rs,
-        &x,
-    )
-    .await?;
+    assert!(NC * input_shares.len() < 120_000);
```

Comment:
There is an argument that defensive coding means this property should be checked here (within `convert_to_fp25519`). But I also think it would be reasonable to say that `convert_to_fp25519` isn't a general purpose routine, so doesn't have to be so aggressive about checking its inputs, and instead of checking this property here, just put the comment about the restriction alongside the assignment of `PROOF_CHUNK`.